src/components/BlogForm/AddBlog.tsx

```diff
@@ -0,0 +1,178 @@
+
+  const onSubmit = (data: addFormData) => {
+    if (!heroPhoto) {
+      return enqueueSnackbar("No here image selected.", { variant: "error" });
+    }
+    let imageId;
+    let blogSlug;
+
+    blogStorage(heroPhoto, data.title).then((res) => {
+      ({ imageId, blogSlug } = res);
+      postBlog(blogSlug, {
+        ...data,
+        author: user.name,
+        imageId: imageId,
+        slug: blogSlug,
+        content: content,
+      })
+        .then(() => enqueueSnackbar("Blog card has been added"))
+        .catch((e) =>
+          enqueueSnackbar(`Error adding the blog card: ${e.message}`, {
+            variant: "error",
+          })
+        )
+        .finally(() => {
+          reset();
+          setHeroPhoto(null);
+          setContent("");
+        });
+    });
+  };
+
+  return (
+    <Container maxWidth="md">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Stack direction={"column"} spacing={2} py={5}>
+          <Controller
+            name="title"
+            control={control}
+            rules={{ required: true }}
+            render={({
+              field: { onChange, value },
+              fieldState: { error },
+            }: any) => (
+              <TextField
+                onChange={onChange}
+                value={value}
+                label={"Blog Title"}
+                error={!!error}
+                helperText={!!error && "A title is required"}
+                size="small"
+              />
+            )}
+          />
+          <Controller
+            name="tags"
+            control={control}
+            rules={{ required: true }}
+            render={({
+              field: { onChange, value },
+              fieldState: { error },
+            }: any) => (
+              <TextField
+                label="Tags"
+                size="small"
+                select
+                fullWidth
+                value={value}
+                onChange={onChange}
+                error={!!error}
+                helperText={!!error && "A Tag is required"}
+              >
+                {tagOptions.map((option) => (
+                  <MenuItem key={option} value={option}>
+                    {option}
+                  </MenuItem>
+                ))}
+              </TextField>
+            )}
+          />
+
+          <Controller
+            name="summary"
+            control={control}
+            rules={{ required: true }}
+            render={({
+              field: { onChange, value },
+              fieldState: { error },
+            }: any) => (
+              <TextField
+                label={"Summary"}
+                onChange={onChange}
+                multiline
+                error={!!error}
+                helperText={!!error && "A Summary is required"}
+                value={value}
+                size="small"
+              />
+            )}
+          />
+
+          <TextField type={"file"} size="small" onChange={handleImageChange} />
```

Comment:
is there a way users can delete / change the image they picked?

Reply:
just click browse and select a new image to change but cant delete

Reply:
ok, for next PR consider the case that the user accidentally uploads incorrect files or data, how would they easily and intuitively correct that mistake?

Reply:
i wanted to add a turnery opp that when a image is uploaded it would show on the screen ygm

Reply:
ye that would be clean, you can access the name of the file through `file.name` i think and display that as well